segmentstore/server/src/main/java/io/pravega/segmentstore/server/reading/ContainerReadIndex.java

```diff
@@ -199,11 +199,11 @@ public void triggerFutureReads(Collection<Long> streamSegmentIds) {
                 // It is possible that between the time we got the pointer to the StreamSegmentReadIndex and when we got
                 // to invoking triggerFutureReads, the StreamSegmentReadIndex has already been closed. If this is the case,
                 // ignore the error.
-                // This is possible in the following scenario: for a Transaction, we have an Append, followed by a Merge;
-                // the Append makes this index eligible for triggering future reads, and the Merge (once committed to Storage)
+                // This is possible in the following scenario: for a Transaction, we have an Append/Seal, followed by a Merge;
+                // the Append/Seal makes this index eligible for triggering future reads, and the Merge (once committed to Storage)
                 // will close it. If the StorageWriter is sufficiently fast in comparison to the OperationProcessor callbacks
                 // (which could be the case for in-memory unit tests), it may trigger this condition.
-                if (getIndex(segmentId) == null) {
+                if (getIndex(segmentId) != null) {
```

Comment:
I agree, and I assume this change is innocent. But I am worried that there are whole code paths that we can't add tests for. I get not catching the problem the first time, but perhaps it's worth taking some time to work out if there is a way to refactor things here so they can be more easily tested.